x-pack/plugins/security_solution/public/timelines/components/timeline/query_tab_content/index.tsx

```diff
@@ -247,10 +251,29 @@ export const QueryTabContentComponent: React.FC<Props> = ({
     timerangeKind,
   });
 
+  const handleOnEventClosed = useCallback(() => {
+    onEventClosed({ timelineId, event: {} });
+    setShowEventDetailsColumn(false);
+
+    if (timelineId === TimelineId.active) {
+      activeTimeline.toggleExpandedEvent({
+        eventId: expandedEvent.eventId!,
+        indexName: expandedEvent.indexName!,
+        loading: false,
+      });
+    }
+  }, [timelineId, onEventClosed, expandedEvent.eventId, expandedEvent.indexName]);
+
   useEffect(() => {
     setIsTimelineLoading({ id: timelineId, isLoading: isQueryLoading || loadingSourcerer });
   }, [loadingSourcerer, timelineId, isQueryLoading, setIsTimelineLoading]);
 
+  useEffect(() => {
+    if (!events || findIndex((e) => e._id === expandedEvent.eventId, events) < 0) {
```

Comment:
I think we could replace `findIndex` with `some` (or anything else that will return us the boolean if the item exists on the list, so we can avoid comparing it to the 0)